src/entities/post/api/api.ts

```diff
@@ -0,0 +1,55 @@
+import { http } from "@shared/lib"
+import type { Tag, Post } from "../model"
+import type { PaginationParams } from "@shared/types/pagination"
+
+export type CreatePostRequest = Pick<Post, "title" | "body" | "userId">
+export type UpdatePostRequest = Partial<Post>
+
+export interface PostsResponse {
+  posts: Post[]
+  total: number
+  skip: number
+  limit: number
+}
+
+export type FetchPostsParams = PaginationParams
+
+export interface FetchPostsByTagParams extends FetchPostsParams {
+  tag: string
+}
+
+export interface FetchPostsBySearchParams extends FetchPostsParams {
+  search?: string
+}
+
+export const postApi = {
+  getPosts: async (params: FetchPostsParams): Promise<PostsResponse> => {
+    return http.get<PostsResponse>("/posts", { params })
+  },
```

Comment:
소소한데, http.get 에 generic 으로 반환 타입이 명시되어 있으니까 래핑한 함수는 반환 타입을 명시하지 않아도 되지 않아도 알아서 잘 추론되지 않을까 싶습니다.


```suggestion
  getPosts: async (params: FetchPostsParams) => {
    return http.get<PostsResponse>("/posts", { params })
  },
```

Reply:
http.get 함수는 정말 좋은 shared 함수인 것 같아요